source/blink.hpp

```diff
@@ -44,16 +46,30 @@ namespace blink
 		}
 
 	private:
+		struct free_delete
+		{
+			void operator()(void *x) { free(x); }
+		};
+		struct notification_info {
+			std::shared_ptr<FILE_NOTIFY_INFORMATION> p_info;
```

Comment:
I'd prefer this to be a `std::vector<BYTE>` instead, so you don't need the `free_delete` hack and C-style `malloc`/`free`. And then just reinterpret that in the for loop as before (`for (auto info = reinterpret_cast<FILE_NOTIFY_INFORMATION *>(notification_infos[dir_index].p_info.data()); ...`).

Reply:
Thanks. Using std::vector<BYTE> and member initializer now.